lib/chef/provisioning/aws_driver/driver.rb

```diff
@@ -893,25 +896,133 @@ def default_ami_for_region(region, criteria = {})
+      if machine_spec.reference[:winrm_ssl_cert]
+        FileUtils.mkdir_p(Chef::Config.trusted_certs_dir)
+        filename = File.join(Chef::Config.trusted_certs_dir, "#{machine_spec.name}.crt")
+        if File.exists?(filename)
+          Chef::Log.warn("Existing cert for #{winrm_subject} in #{filename}")
+        else
+          Chef::Log.warn("Adding certificate for #{winrm_subject} in #{filename}")
+          File.open(filename, File::CREAT|File::TRUNC|File::RDWR, 0644) do |f|
+            f.print(machine_spec.reference[:winrm_ssl_cert])
+          end
+        end
+        winrm_options[:ca_trust_path] = filename
+      end
       Chef::Provisioning::Transport::WinRM.new("#{endpoint}", type, winrm_options, {})
+      binding.pry
+      # in order for ssl to work, we need to tell ssl that it's ok that ssl_subject doesn't
+      # match the ip
+      #[19] pry(#<Chef::Provisioning::AWSDriver::Driver>)> endpoint.split('/')[2].split(':').first
+      #  => "10.113.70.104"
+      #  [20] pry(#<Chef::Provisioning::AWSDriver::Driver>)> machine_spec.reference[:winrm_ssl_subject]
+      #    => "IP-0A714668"
+      #
+      # [1] pry(#<Chef::Provisioning::AWSDriver::Driver>)> Chef::Provisioning::Transport::WinRM.new("#{endpoint}", type, winrm_options, {}).execute('hostname')
+      # OpenSSL::SSL::SSLError: SSL_connect returned=1 errno=0 state=error: certificate verify failed
+      # from /home/hh/.rvm/gems/ruby-2.2.0/gems/httpclient-2.6.0.1/lib/httpclient/session.rb:307:in `connect'
+      # [2] pry(#<Chef::Provisioning::AWSDriver::Driver>)> wtf!
+      # Exception: OpenSSL::SSL::SSLError: SSL_connect returned=1 errno=0 state=error: certificate verify failed
+      # --
+      # 0: /home/hh/.rvm/gems/ruby-2.2.0/gems/httpclient-2.6.0.1/lib/httpclient/session.rb:307:in `connect'
+      # 1: /home/hh/.rvm/gems/ruby-2.2.0/gems/httpclient-2.6.0.1/lib/httpclient/session.rb:307:in `ssl_connect'
+      # 2: /home/hh/.rvm/gems/ruby-2.2.0/gems/httpclient-2.6.0.1/lib/httpclient/session.rb:755:in `block in connect'
+      # 3: /home/hh/.rvm/rubies/ruby-2.2.0/lib/ruby/2.2.0/timeout.rb:89:in `block in timeout'
+      # 4: /home/hh/.rvm/rubies/ruby-2.2.0/lib/ruby/2.2.0/timeout.rb:99:in `call'
+      # 5: /home/hh/.rvm/rubies/ruby-2.2.0/lib/ruby/2.2.0/timeout.rb:99:in `timeout'
+      # 6: /home/hh/.rvm/rubies/ruby-2.2.0/lib/ruby/2.2.0/timeout.rb:125:in `timeout'
+      # 7: /home/hh/.rvm/gems/ruby-2.2.0/gems/httpclient-2.6.0.1/lib/httpclient/session.rb:746:in `connect'
+      # 8: /home/hh/.rvm/gems/ruby-2.2.0/gems/httpclient-2.6.0.1/lib/httpclient/session.rb:612:in `query'
+      # 9: /home/hh/.rvm/gems/ruby-2.2.0/gems/httpclient-2.6.0.1/lib/httpclient/session.rb:164:in `query'
+      # [3] pry(#<Chef::Provisioning::AWSDriver::Driver>)> winrm_options
+      # => {:user=>"Administrator",
+      #  :pass=>"(xntd8f=-HNnuJ3",
+      #   :disable_sspi=>false,
+      #    :basic_auth_only=>false,
+      #     :no_ssl_peer_verification=>false,
```

Comment:
@mwrock 
The purpose here is to make it easy to copy the already existing self signed cert generated by windows on first boot for rdp, and reuse it for winrm. The Subject CN of that cert does not contain a valid fqdn, it's usually WIN-RANDOM or IP-HEXIPV4.

I'm trying to duplicate the utility of ssh where we verify the fingerprint of the connection to a particular ip (and verify it). See http://www.rackspace.com/knowledge_center/article/rackspace-cloud-essentials-checking-a-server%E2%80%99s-ssh-host-fingerprint-with-the-web-console

I'm looking for a way to explicitly pass self-signed ssl certificates per https connection. The only reason it's failing right now is the Certificate CN doesn't match the ip we attempt to connect to.

If we aren't able to actually verify self-signed ssl certs and bind them to a particular winrm target it kinda makes self-signed certs useless unless we regenerate them to match their IP or a dns name that reaches them... which is not easy to do in ec2 user_data.

I'm currently using the following:

```powershell

winrm quickconfig -q                                                                                                                           
winrm set winrm/config/winrs '@{MaxMemoryPerShellMB="300"}'                                                                                    
winrm set winrm/config '@{MaxTimeoutms="1800000"}'                                                                                             
                                                                                                                                               
netsh advfirewall firewall add rule name="WinRM 5986" protocol=TCP dir=in localport=5986 action=allow                                          
                                                                                                                                               
$SourceStoreScope = 'LocalMachine'                                                                                                             
$SourceStorename = 'Remote Desktop'                                                                                                            
                                                                                                                                               
$SourceStore = New-Object  -TypeName System.Security.Cryptography.X509Certificates.X509Store  -ArgumentList $SourceStorename, $SourceStoreScop\
e                                                                                                                                              
$SourceStore.Open([System.Security.Cryptography.X509Certificates.OpenFlags]::ReadOnly)                                                         
                                                                                                                                               
$cert = $SourceStore.Certificates | Where-Object  -FilterScript {                                                                              
    $_.subject -like '*'                                                                                                                       
}                                                                                                                                              
                                                                                                                                               
$DestStoreScope = 'LocalMachine'                                                                                                               
$DestStoreName = 'My'                                                                                                                          
                                                                                                                                               
$DestStore = New-Object  -TypeName System.Security.Cryptography.X509Certificates.X509Store  -ArgumentList $DestStoreName, $DestStoreScope      
$DestStore.Open([System.Security.Cryptography.X509Certificates.OpenFlags]::ReadWrite)                                                          
$DestStore.Add($cert)                                                                                                                          
                                                                                                                                               
$SourceStore.Close()                                                                                                                           
$DestStore.Close()                                                                                                                             
                                                                                                                                               
winrm create winrm/config/listener?Address=*+Transport=HTTPS  `@`{Hostname=`"($certId)`"`;CertificateThumbprint=`"($cert.Thumbprint)`"`}       
                                                                                                                                               
net stop winrm                                                                                                                                 
sc config winrm start=auto                                                                                                                     
net start winrm 
```